cesql/spec.md

```diff
@@ -56,7 +56,10 @@ producer, or in an intermediary, and it can be implemented using any technology
 The CloudEvents Expression Language assumes the input always includes, but is not limited to, a single valid and
 type-checked CloudEvent instance. An expression MUST NOT mutate the value of the input CloudEvent instance, nor any of
 the other input values. The evaluation of an expression observes the concept of [referential
-transparency][referential-transparency-wiki]. The output of a CESQL expression evaluation is always a _boolean_, an _integer_ or a _string_, and it might include an error.
+transparency][referential-transparency-wiki]. The primary output of a CESQL expression evaluation is always a _boolean_, an _integer_ or a _string_.
```

Comment:
Just curious, do you know why the reference to "referential transparency" is here? I'm just wondering what people should do with that info? If the sentence were removed would it hurt the spec or change the impls?

Reply:
Oh as far as I know the point of this is that there are no side effects in a CESQL statement - every function/operator/subexpression can be replaced with the output of that function/operator/subexpression and you would get the same result

Reply:
Thinking about this, I don't think it necessarily hurts the spec, but I find that it does give some clarity about the independence of parts of an expression. I'll try adding another sentence explaining the referential transparency